autobot/telegram/parser.py

```diff
@@ -238,16 +238,459 @@ def _entity_parser (entity:dict):
         return msg_obj
 
     # methods to parse games objects
+    def _parse_game (self,key:str,val:dict) -> Game|GameHighScore:
+        game_obj = None
+
+        if (key == "game"):
+            game_obj = Game()
+            for k,v in val.items():
+                match k:
+                    case "photo":
+                        pht = self._parse_photosize(k,v)
+                        setattr(game_obj,k,pht)
+                    case "animation":
+                        anim = self._parse_animation(k,v)
+                        setattr(game_obj,k,anim)
+                    case _:
+                        setattr(game_obj,k,v)
+
+        elif (key == "game_high_score"):
+            game_obj = GameHighScore()
+            for k,v in val.items():
+                setattr(game_obj,k,v)
+
+        return game_obj
     
     
     # methods to parse inline objects
+    def _parse_inlinequeryresults (self,key:str,val:dict) -> InlineQueryResult|InlineQueryResultArticle|InlineQueryResultAudio|InlineQueryResultCachedAudio|InlineQueryResultCachedDocument|InlineQueryResultCachedGif|InlineQueryResultCachedMpeg4Gif|InlineQueryResultCachedPhoto|InlineQueryResultCachedSticker|InlineQueryResultCachedVideo|InlineQueryResultCachedVoice|InlineQueryResultContact|InlineQueryResultDocument|InlineQueryResultGame|InlineQueryResultGif|InlineQueryResultLocation|InlineQueryResultMpeg4Gif|InlineQueryResultPhoto|InlineQueryResultVenue|InlineQueryResultVideo|InlineQueryResultVoice:
+        inline_obj = None
+
+        if (key == "inline_query_result"):
+            inline_obj = InlineQueryResult()
+            for k,v in val.items():
+                match k:
+                    case "photo_url"|"thumb_url"|"gif_url"|"mpeg4_url"|"video_url"|"audio_url"|"voice_url"|"document_url"|"sticker_url"|"photo_file_id"|"gif_file_id"|"mpeg4_file_id"|"video_file_id"|"audio_file_id"|"voice_file_id"|"document_file_id"|"sticker_file_id":
+                        setattr(inline_obj,k,v)
+                    case "input_message_content":
+                        imc = self._parse_inputmessagecontent(k,v)
+                        setattr(inline_obj,k,imc)
+                    case "reply_markup":
+                        rep = self._parse_inline(k,v)
```

Comment:
I still see _parse_inline on line 279, I hope you'll rename it to the actual function responsible for the parsing

Reply:
`_parse_inlinequery` ??

Reply:
@kallyas I'm not sure which object "reply_markup" maps to, but you should use a function name that describes the responsibility of the function and which objects it's responsible for parsing. So if _parse_inlinequery seems suitable, we can go with it.

Reply:
It's an `InlineKeyboardMarkup`  so I'm thinking if we can use `_parse_inlinekeyboard` or `_parse_inlinekeyboardobjects`

Reply:
@kallyas I think `_parse_inlinekeyboardobjects` should work there then. I see you've made that change. I can't seem to find any other issues, so you can go ahead and merge this. We shall test to see if there are any errors